### ISSUE TITLE
Color type option fires on next option change

### ISSUE THREAD

**Issue by danyj** [themeum/kirki#206]
See this http://screencast.com/t/6RpmklMKM4FD,  possibly connected to  #187

**Comment by aristath** [comment]
When you change the color value and it's not refreshed on the site preview, does the "save" button on the top allow you to save the options?
If not, then perhaps it doesn't even detect that the control value has changed...
If it does, then perhaps it's just the refresh that doesn't get triggered.

Are you by any chance using `'transport' => 'postMessage'` on that control?
If you are, have you set the `js_vars` argument as documented here? http://kirki.org/#configuration (yeah, I know the docs need some work... but that should give you an idea.

If you're not using that, can you try using it and let me know if it makes a difference?

**Comment by aristath** [comment]
Sorry, wrong link on the docs. This is the right one: http://kirki.org/#js_vars

**Comment by danyj** [comment]
The save button goes from save to Save & Publish once I click inside the rainbow

using your default demo dist

this is that option  and yes it has transport

```
$controls[] = array(
    'type'        => 'color',
    'setting'     => 'color_demo',
    'label'       => __( 'This is the label', 'kirki' ),
    'description' => __( 'This is the control description', 'kirki' ),
    'help'        => __( 'This is some extra help. You can use this to add some additional instructions for users. The main description should go in the "description" of the field, this is only to be used for help tips.', 'kirki' ),
    'section'     => 'color_section',
    'default'     => '#0088cc',
    'priority'    => 10,
    'output'      => array(
        array(
            'element'  => 'a, a:visited',
            'property' => 'color',
            'units'    => ' !important'
        ),
        array(
            'element'  => '#content',
            'property' => 'border-color'
        ),
    ),
    'transport' => 'postMessage',
    'js_vars'   => array(
        array(
            'element'  => 'a, a:visited',
            'function' => 'css',
            'property' => 'color',
        ),
        array(
            'element'  => '#content',
            'function' => 'css',
            'property' => 'background-color',
        ),
    )
);
```

yes I see if transport is removed it works. makes sense now . closing

**Comment by saltnlight** [comment]
I'm struggling with this issue. Changing things on the color picker only fires the 'wp.customize' script on the second click on the 'preset colors' on the left of the picker. Dragging the 'rainbow' around and moving the sliders is firing the 'wp.customize' script fine.

I've noticed that clicking on a preset is changing the value in the input field but it's still selected and the text cursor is still in there so it's not firing the wp.customize script. Should this be the case or am I doing something wrong?

I've tried both using the kirki js_vars and writing my own postMessage and I get the same issue with both. I'm also hitting this issue with presets too. As soon as I change to a new multi-option preset all of my colors are delayed by one preset causing chaos!

Hopefully this gif will help...http://www.saltnlight.co.uk/demosite/files/2015/11/Error.gif

Notice how editing the opacity manually in the input field will fire the script but just changing the bracket at the end won't do it?! Weird. Pressing Enter doesn't fire it either which I was expecting.
